x-pack/plugins/security_solution/public/common/lib/kuery/index.ts

```diff
@@ -199,22 +200,26 @@ export const isDataProviderEmpty = (dataProviders: DataProvider[]) => {
   return isEmpty(dataProviders) || isEmpty(dataProviders.filter((d) => d.enabled === true));
 };
 
+export const dataViewSpecToViewBase = (dataViewSpec?: DataViewSpec): DataViewBase => {
+  return { title: dataViewSpec?.title || '', fields: Object.values(dataViewSpec?.fields || {}) };
+};
+
 export const convertToBuildEsQuery = ({
   config,
   indexPattern,
   queries,
   filters,
 }: {
   config: EsQueryConfig;
-  indexPattern: DataViewBase | undefined;
+  indexPattern: DataViewSpec | undefined;
   queries: Query[];
   filters: Filter[];
 }): [string, undefined] | [undefined, Error] => {
   try {
     return [
       JSON.stringify(
         buildEsQuery(
-          indexPattern,
+          dataViewSpecToViewBase(indexPattern),
```

Comment:
Same thing here with the `indexPattern` to `dataViewSpec`, even the util function to change on line 222 calls it that way. This function is used a bit more extensively than the last one but I still think the change would be worth it going forward.